doc/user_guide/psyir.rst

```diff
@@ -100,6 +100,55 @@ subclassed into a kernel-layer ``KernelSchedule``. In addition to
 ``BinaryOperation`` and ``NaryOperation``.
 
 
+Node Descriptions
+=================
+
+The Range node
+--------------
+
+The ``Range`` node is used to capture a range of integers via ``start``,
+``stop`` and ``step`` values. For example, ``start=2``, ``stop=6`` and
+``step=2`` indicates the values ``2``, ``4`` and ``6``.
+
+At the moment the only valid use of ``Range`` in the PSyIR is to
+describe a set of accesses to an Array dimension (so-called array
+notation in Fortran). Therefore, the parent of a ``Range`` node should
+only be an ``Array`` node.
+
+The ``Range`` node has three children nodes, the first child captures
+the start of the range, the second child captures the end of the range
+and the third captures the step within the range.
+
+The nodes for each of the children must return an integer. Potentially
+valid nodes are therefore ``Literal``, ``Reference``, ``Operation``
+and ``CodeBlock``.
+
+A common use case is to want to specify all the elements of the array
```

Comment:
"...elements of a given array dimension without knowing the extent of that dimension. In the PSyIR..."

Reply:
Better, thanks.